p2pool/src/sharechain/in_memory.rs

```diff
@@ -377,9 +377,11 @@ impl InMemoryShareChain {
         None
     }
 
-    fn get_calculate_and_cache_hashmap_of_tip_shares(
+    fn calculate_and_cache_hashmap_shares(
         &self,
         p2_chain: &mut RwLockWriteGuard<'_, P2Chain<LmdbBlockStorage>>,
+        calculating_height: u64,
+        prev_hash: &FixedHash,
```

Comment:
This is not always for the previous hash
```suggestion
        calculating_hash: &FixedHash,
```